pkgs/development/python-modules/ratarmount/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, isPy3k
+, fusepy
+, ratarmountcore
+}:
+
+buildPythonPackage rec {
+  pname = "ratarmount";
+  version = "0.14.0";
+
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3fea7487e2ae3acba73ec5db470c73021af55dc12a3234311de1e6036f7ea434";
+  };
+
+  propagatedBuildInputs = [ ratarmountcore ];
+
```

Comment:
That being said, i'm not convinced the tests properly ran:

  checkPhase = ''                                                                                                                                                                                                                   
    runHook preCheck                                                                                                                                                                                                                
    python tests/tests.py                                                                                                                                                                                                           
    runHook postCheck                                                                                                                                                                                                               
  '';

Reply:
Ah right, forgot about these short tests. The one taking an hour mentioned in my other comment are inside tests/runtests.sh

> , i'm not convinced the tests properly ran

Looks ok to me, I see output from the tests:

```pascal
ewriting #!/nix/store/mnqzpjm53jk1mfiyj189ri9qfk5wmjqf-python3-3.10.12/bin/python3.10 to #!/nix/store/mnqzpjm53jk1mfiyj189ri9qfk5wmjqf-python3-3.10.12
wrapping `/nix/store/nid6zpjjnwm443184h85nkwv0v1snqc3-python3.10-ratarmount-0.14.0/bin/ratarmount'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests

Test creating and using an index with .tar.gz files with SQLiteIndexedTar
Created temp tar: /build/tmpgjerw2iz.tar.gz

== Test with file paths ==
[Info] Detected compression gz for file object: <_io.BufferedReader name='/build/tmpgjerw2iz.tar.gz'>
[Info] Do not reopen with rapidgzip backend because:
[Info]  - is too small to qualify for parallel decompression.
[Info] Undid gz file compression by using: IndexedGzipFile
Creating new SQLite index database at /build/tmpk1tbnzlx.sqlite
Creating offset dictionary for /build/tmpgjerw2iz.tar.gz ...
Resorting files by path ...
Creating offset dictionary for /build/tmpgjerw2iz.tar.gz took 0.00s
[Info] Could not load GZip Block offset data. Will create it from scratch.
Writing out TAR index to /build/tmpk1tbnzlx.sqlite took 0s and is sized 24576 B
[Info] Detected compression gz for file object: <_io.BufferedReader name='/build/tmpgjerw2iz.tar.gz'>
[Info] Do not reopen with rapidgzip backend because:
[Info]  - is too small to qualify for parallel decompression.
[Info] Undid gz file compression by using: IndexedGzipFile
Successfully loaded offset dictionary from /build/tmpk1tbnzlx.sqlite
Loading gzip block offsets took 0.00s

== Test with file objects ==
[Info] Detected compression gz for file object: <_io.BufferedReader name='/build/tmpgjerw2iz.tar.gz'>
[Info] Do not reopen with rapidgzip backend because:
[Info]  - is too small to qualify for parallel decompression.
[Info] Undid gz file compression by using: IndexedGzipFile
Creating new SQLite index database at /build/tmpk1tbnzlx.sqlite
Creating offset dictionary for tarFileName ...
Resorting files by path ...
Creating offset dictionary for tarFileName took 0.00s
[Info] Could not load GZip Block offset data. Will create it from scratch.
Writing out TAR index to /build/tmpk1tbnzlx.sqlite took 0s and is sized 24576 B
[Info] Detected compression gz for file object: <_io.BufferedReader name='/build/tmpgjerw2iz.tar.gz'>
[Info] Do not reopen with rapidgzip backend because:
[Info]  - is too small to qualify for parallel decompression.
[Info] Undid gz file compression by using: IndexedGzipFile
Successfully loaded offset dictionary from /build/tmpk1tbnzlx.sqlite
Loading gzip block offsets took 0.00s

== Test with file objects with no fileno ==
[Info] Detected compression gz for file object: <_io.BytesIO object at 0x7ffff7512c00>
[Info] Do not reopen with rapidgzip backend because:
[Info]  - the file to open is a recursive file, which limits the usability of 
[Info]    parallel decompression.
[Info]  - is too small to qualify for parallel decompression.
[Info] Undid gz file compression by using: IndexedGzipFile
Creating new SQLite index database at /build/tmpk1tbnzlx.sqlite
Creating offset dictionary for tarFileName ...
Resorting files by path ...
Creating offset dictionary for tarFileName took 0.00s
[Info] Could not load GZip Block offset data. Will create it from scratch.
Writing out TAR index to /build/tmpk1tbnzlx.sqlite took 0s and is sized 24576 B
[Info] Detected compression gz for file object: <_io.BytesIO object at 0x7ffff7512c00>
[Info] Do not reopen with rapidgzip backend because:
[Info]  - the file to open is a recursive file, which limits the usability of 
[Info]    parallel decompression.
[Info]  - is too small to qualify for parallel decompression.
[Info] Undid gz file compression by using: IndexedGzipFile
Successfully loaded offset dictionary from /build/tmpk1tbnzlx.sqlite
Loading gzip block offsets took 0.00s
pythonCatchConflictsPhase
/nix/store/91qrq1mz3d6qz4lv2zirswb3z5qr538l-catch_conflicts.py:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: ratarmount
/nix/store/nid6zpjjnwm443184h85nkwv0v1snqc3-python3.10-ratarmount-0.14.0
```

They are quite verbose because they call ratarmount with printDebug=3.